src/fsharp/FSharp.Core/seq.fs

```diff
@@ -1385,9 +1385,10 @@ namespace Microsoft.FSharp.Collections
             | :? ('T[]) as a -> 
                 if a.Length = 0 then invalidArg "source" LanguagePrimitives.ErrorStrings.InputSequenceEmptyString
                 else a.[a.Length - 1]
-            | :? IList<'T> as a -> 
+            | :? ('T IList) as a -> //ResizeArray and other collections 
                 if a.Count = 0 then invalidArg "source" LanguagePrimitives.ErrorStrings.InputSequenceEmptyString
                 else a.[a.Count - 1]
+            | :? list<'T> as a -> List.last a                
```

Comment:
I'm not sure that using `List.last` is correct, as it may throw a (slightly) different error than the error for `InputSequenceEmptyString`. After all, the user took the `Seq.last` function and not the `List.last`, but I don't know what the policy is (@dsyme?) when it comes to such subtle differences.

As an alternative, consider `match a with [] -> invalidarg ... | _ -> List.last a`.

Also, common coding practice is `'T list` I believe ;).

Reply:
fixed

Reply:
List.last is not available due to compilation order.
suggested fix https://github.com/dotnet/fsharp/pull/7765/commits/b329e23c543b41b36143fafc77187acc58585471

Reply:
List.last is not available due to compilation order.
I copied it.
@dsyme will this be inlined because its tail recursive ? Or would it be better to have this recursive function privately in the main scope.
https://github.com/dotnet/fsharp/pull/7765/commits/ef43d0ed1d91cd05ac6671d5c7ed13f780f69a82